pkg/sql/sqlstats/insights/integration/integration_test: TestInsightsIntegrationForContention failed

pkg/sql/sqlstats/insights/integration/integration_test.TestInsightsIntegrationForContention [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10871800?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10871800?buildTab=artifacts#/) on master @ [92d5d0e329e1ed2d923c5ff2116fac9813e4dac5](https://github.com/cockroachdb/cockroach/commits/92d5d0e329e1ed2d923c5ff2116fac9813e4dac5):


```
=== RUN   TestInsightsIntegrationForContention
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/8db7e2fbdb77cb910d7e66552998132e/logTestInsightsIntegrationForContention1948877142
    test_log_scope.go:81: use -show-logs to present logs inline
    insights_test.go:566: condition failed to evaluate within 5s: contention time is 0.111568 should be greater than .2 since block is delayed by .5 seconds
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/8db7e2fbdb77cb910d7e66552998132e/logTestInsightsIntegrationForContention1948877142
--- FAIL: TestInsightsIntegrationForContention (7.77s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cluster-observability
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInsightsIntegrationForContention.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-29647

pkg/sql/sqlstats/insights/integration/integration_test.TestInsightsIntegrationForContention [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10913258?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10913258?buildTab=artifacts#/) on master @ [d4640566f77ddcf03317ec8ccfbea2aa1568601d](https://github.com/cockroachdb/cockroach/commits/d4640566f77ddcf03317ec8ccfbea2aa1568601d):


```
=== RUN   TestInsightsIntegrationForContention
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/8db7e2fbdb77cb910d7e66552998132e/logTestInsightsIntegrationForContention3619859221
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:98: 
        Test server was configured to route SQL queries to a secondary tenant (virtual cluster).
        If you are only seeing a test failure when this message appears, there may be a problem
        specific to cluster virtualization or multi-tenancy.
        
        To investigate, consider using "COCKROACH_TEST_TENANT=true" to force-enable just
        the secondary tenant in all runs (or, alternatively, "false" to force-disable), or use
        "COCKROACH_INTERNAL_DISABLE_METAMORPHIC_TESTING=false" to disable all random test variables altogether.
    insights_test.go:566: condition failed to evaluate within 5s: contention time is 0.119277 should be greater than .2 since block is delayed by .5 seconds
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/8db7e2fbdb77cb910d7e66552998132e/logTestInsightsIntegrationForContention3619859221
--- FAIL: TestInsightsIntegrationForContention (7.84s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInsightsIntegrationForContention.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/sqlstats/insights/integration/insights_test.go
+++ b/pkg/sql/sqlstats/insights/integration/insights_test.go
@@ -572,11 +572,11 @@ func TestInsightsIntegrationForContention(t *testing.T) {
 		txn_contention.database_name,
 		txn_contention.table_name,
 		txn_contention.index_name,
-		txn_contention.waiting_txn_fingerprint_id
+		encode(txn_contention.waiting_txn_fingerprint_id, 'hex') AS waiting_txn_fingerprint_id
 		FROM crdb_internal.cluster_execution_insights insight
 		left join crdb_internal.transaction_contention_events txn_contention on  insight.stmt_id = txn_contention.waiting_stmt_id
 																		 where query like 'UPDATE t SET s =%'
-		group by query, insight.contention, txn_contention.schema_name, txn_contention.database_name, txn_contention.table_name, txn_contention.index_name, waiting_txn_fingerprint_id;`)
+		group by query, insight.contention, txn_contention.schema_name, txn_contention.database_name, txn_contention.table_name, txn_contention.index_name, txn_contention.waiting_txn_fingerprint_id;`)
 		if err != nil {
 			return err
 		}
@@ -589,15 +589,18 @@ func TestInsightsIntegrationForContention(t *testing.T) {
 			}
 
 			var totalContentionFromQueryMs, contentionFromEventMs float64
-			var queryText, schemaName, dbName, tableName, indexName string
-			var blockingTxnFingerprintID gosql.NullString
-			err = rows.Scan(&queryText, &totalContentionFromQueryMs, &contentionFromEventMs, &schemaName, &dbName, &tableName, &indexName, &blockingTxnFingerprintID)
+			var queryText, schemaName, dbName, tableName, indexName, waitingTxnFingerprintID string
+			err = rows.Scan(&queryText, &totalContentionFromQueryMs, &contentionFromEventMs, &schemaName, &dbName, &tableName, &indexName, &waitingTxnFingerprintID)
 			if err != nil {
 				return err
 			}
 
-			if totalContentionFromQueryMs < .2 {
-				return fmt.Errorf("contention time is %f should be greater than .2 since block is delayed by .5 seconds", totalContentionFromQueryMs)
+			if totalContentionFromQueryMs <= 0 {
+				return fmt.Errorf("contention time is %f must be greater than 0", totalContentionFromQueryMs)
+			}
+
+			if totalContentionFromQueryMs > 60*1000 {
+				return fmt.Errorf("contention time must be less than 1 minute:  %f", totalContentionFromQueryMs)
 			}
 
 			diff := totalContentionFromQueryMs - contentionFromEventMs
@@ -621,8 +624,8 @@ func TestInsightsIntegrationForContention(t *testing.T) {
 				return fmt.Errorf("index names do not match 't_pkey', %s", indexName)
 			}
 
-			if !blockingTxnFingerprintID.Valid {
-				return fmt.Errorf("blockingTxnFingerprintId is null")
+			if waitingTxnFingerprintID == "0000000000000000" || waitingTxnFingerprintID == "" {
+				return fmt.Errorf("waitingTxnFingerprintID is default value: %s", waitingTxnFingerprintID)
 			}
 		}
 

